yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -156,87 +185,68 @@ func packAndSendAssessMigrationPayload(status string, errMsg string) {
 		}),
 	}
 
-	unsupportedDatatypesList := lo.Map(assessmentReport.UnsupportedDataTypes, func(datatype utils.TableColumnsDataTypes, _ int) string {
-		return datatype.DataType
-	})
+	explanation, err := buildMigrationComplexityExplanation(source.DBType, assessmentReport, "")
```

Comment:
Why are we building again? It should already be available in the assessment report, right?

Reply:
the reason is, we will this twice... for json and html reports both, at the time of report generation.
whatever happens later will be stored in the struct's field.

So just to be on safer side i am building it again.

Reply:
Ah, right. See, here's another reason why having different values for the same field (depending on report type) is not a good idea. `AssessmentReport.MigrationComplexityExplanation` should be properly defined in our struct (without including any html/json specific logic). 
- If we want any HTML specific logic, it should be in the template. 
- If we want any json-specific logic, it should be in a custom marshaller.
- If we want any callhome specific logic, it should be in the callhome layer (this function)

Pls add a comment here explaining why we're re-building. And let's clean this up soon

Reply:
yeah agree, only limitation which stopped me was that to the template we can pass only one struct.
In the original assessment report struct, i wanted to avoid a new child struct, just for the html report case.

Reply:
Ah I see. It might not be such a bad idea to have a chid struct actually. But alternatively, you could just have all the fields in the struct directly, and then mark them as json:"-" if you don't want them in the json